src/lib/tcti_ldr.c

```diff
@@ -168,49 +168,40 @@ tcti_conf tcti_get_config(void) {
     };
 
     /* no tcti config supplied, get it from env */
-    char *optstr = getenv (TPM2_PKCS11_TCTI);
-    if (!optstr) {
+    char *env_conf_string = getenv(TPM2_PKCS11_TCTI);
+    if (!env_conf_string) {
         /* nothing user supplied, use default */
         return conf;
     }
 
-    char *split = strchr(optstr, ':');
+    char *conf_string = malloc(strlen(env_conf_string) + 1);
+    strcpy(conf_string, env_conf_string);
+
+    char *split = strchr(conf_string, ':');
     if (!split) {
         /* --tcti=device */
-        conf.name = fixup_name(optstr);
-        return conf;
-    }
-
-    /*
-     * If it has a ":", it could be either one of the following:
-     * case A: --tcti=:               --> default name and default (null) config
-     * case B: --tcti=:/dev/foo       --> default name, custom config
-     * case C: --tcti=device:         --> custom name, default (null) config
-     * case D: --tcti=device:/dev/foo --> custom name, custom config
-     */
-
-    split[0] = '\0';
-
-    /* Case A */
-    if (!optstr[0] && !split[1]) {
-        return conf;
-    }
+        conf.name = fixup_name(conf_string);
+    } else {
+        /*
+         * If it has a ":", it could be either one of the following:
+         * case A: --tcti=:               --> default name and default (null) config
+         * case B: --tcti=:/dev/foo       --> default name, custom config
+         * case C: --tcti=device:         --> custom name, default (null) config
+         * case D: --tcti=device:/dev/foo --> custom name, custom config
+         */
+        split[0] = '\0';
 
-    /* Case B */
-    if (!optstr[0]) {
-        conf.opts = &split[1];
-        return conf;
-    }
+        /* Case C, D */
+        if (conf_string[0]) {
```

Comment:
You're stated goal here was to reduce branches, but you added a branch and indentation under an else clause to get to the point of a signgle return. This could be all done with else if's but this refactor should be in a different patch.

Reply:
Agree on the general principle of separating bug fixes from refactoring.